src/dotnet/commands/dotnet-install/dotnet-install-tool/InstallToolCommand.cs

```diff
@@ -43,21 +43,14 @@ public override int Execute()
         {
             var executablePackagePath = new DirectoryPath(new CliFolderPathCalculator().ExecutablePackagesPath);
 
-            var toolConfigurationAndExecutableDirectory = ObtainPackage(executablePackagePath);
-
-            DirectoryPath executable = toolConfigurationAndExecutableDirectory
-                .ExecutableDirectory
-                .WithSubDirectories(
-                    toolConfigurationAndExecutableDirectory
-                        .Configuration
-                        .ToolAssemblyEntryPoint);
+            var toolConfigurationAndExecutablePath = ObtainPackage(executablePackagePath);
 
             var shellShimMaker = new ShellShimMaker(executablePackagePath.Value);
-            var commandName = toolConfigurationAndExecutableDirectory.Configuration.CommandName;
+            var commandName = toolConfigurationAndExecutablePath.Configuration.CommandName;
             shellShimMaker.EnsureCommandNameUniqueness(commandName);
 
             shellShimMaker.CreateShim(
-                executable.Value,
+                toolConfigurationAndExecutablePath.Executable.Value,
```

Comment:
NIT: Should ToolConfigurationAndExecutablePath have a ExecutablePath property, instead of this ugly Executable.Value? I think it would read better.

Reply:
CreateShim should take in a strong type FilePath. However this will conflict with Nate's work. So I will change it later.